os/kernel/sched/sched_addDeadtask.c

```diff
@@ -0,0 +1,255 @@
+/****************************************************************************
+ *
+ * Copyright 2025 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <tinyara/sched.h>
+#include <tinyara/irq.h>
+
+#include <assert.h>
+#include <debug.h>
+#include <queue.h>
+#include <string.h>
+#include "sched/sched.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Type Declarations
+ ****************************************************************************/
+
+struct dead_pid_s {
+    struct dead_pid_s *next;
+    int pid;
+    struct dead_name_s *name_ptr;
+};
+
+typedef struct dead_pid_s dead_pid_t;
+
+struct dead_name_s {
+    struct dead_name_s *next;
+    int pid_cnt;
```

Comment:
minor suggestion, may be naming it as reference count or something along those lines would be better.